### DISCUSSION TITLE
Device Manager width bypass

### DISCUSSION THREAD

**Discussion by adam3278** [GrapesJS/grapesjs#5918] [Q&A]
Hi,
How to turn off editor width changing when device change? I decided to manually (on event) change the width of editor el container, so I need the media width change only. As the result, I set the device width to '' everywhere and editor works perfectly fine now, but the gjs-tools are moving away of the screen (left css prop). How to fix their weird behavior? Thanks in advance.

Device Manager config
```js
deviceManager: {
    devices: [
      {
        id: 'desktop',
        name: 'Desktop',
        width: ''
      },{
        id: 'tablet',
        name: 'Tablet',
        width: '',
        widthMedia: '850px',
      },{
        id: 'mobile',
        name: 'Mobile',
        width: '',
        widthMedia: '500px',
      }
    ]
  },
```

Container width change function
```js
editor.on('device:select', (newDevice, oldDevice) => {
    let contentWrapper = editor.editorView.el?.parentNode;
    switch (newDevice.attributes.id) {
      case 'tablet':
        contentWrapper.setAttribute('style', `width: 850px !important`);
        break;

      case 'mobile':
        contentWrapper.setAttribute('style', `width: 500px !important`);
        break;

      default:
        contentWrapper.style.width = null;
    }
  });
```

**Comment by artf** [comment]
Did you try using [editor.refresh](https://grapesjs.com/docs/api/editor.html#refresh)?
Anyway, having a reproducible demo would help here
